In [2]:
import pandas as pd
import os
import numpy as np

1. quantos rótulos eu troquei x quantos rótulos Adson trocou
2. quantos rótulos nós discordávamos x quantos rótulos nós discordamos agora.
3. gráfico antigo x gráfico atualizado de VP, VN, FP, FN



In [40]:
count_changed_total = 0
test_cases_total = 0
for approach in ["approach1.xlsx", "approach2.xlsx", "approach3.xlsx", "approach4.xlsx", "approach5.xlsx", "random.xlsx"]:
    ann1_revised_dict = pd.read_excel(f"ann1/revised/{approach}", sheet_name=None)
    ann1_divergent_dict = pd.read_excel(f"ann1/{approach}", sheet_name=None)
    try:
        del ann1_revised_dict["templates"]
        del ann1_divergent_dict["templates"]
    except KeyError:
        pass
    
    assert ann1_revised_dict.keys() == ann1_divergent_dict.keys()
    
    count_changed_by_approach = 0
    test_cases_by_approach = 0
    
    for t in ann1_revised_dict:
        revised = ann1_revised_dict[t]['classification']
        not_revised = ann1_divergent_dict[t]['classification']

        count_changed_by_approach += sum(revised != not_revised)
        test_cases_by_approach += len(revised)

    count_changed_total += count_changed_by_approach
    test_cases_total += test_cases_by_approach
    print(f"{approach}:\t{count_changed_by_approach = } from {test_cases_by_approach}")

print(f"{count_changed_total} changed from {test_cases_total}")

approach1.xlsx:	count_changed_by_approach = 51 from 212
approach2.xlsx:	count_changed_by_approach = 11 from 204
approach3.xlsx:	count_changed_by_approach = 13 from 101
approach4.xlsx:	count_changed_by_approach = 16 from 101
approach5.xlsx:	count_changed_by_approach = 27 from 178
random.xlsx:	count_changed_by_approach = 12 from 53
130 changed from 849


- Filipe:
    - approach1.xlsx:	count_changed_by_approach = 51 from 212
    - approach2.xlsx:	count_changed_by_approach = 11 from 204
    - approach3.xlsx:	count_changed_by_approach = 13 from 101
    - approach4.xlsx:	count_changed_by_approach = 16 from 101
    - approach5.xlsx:	count_changed_by_approach = 27 from 178
    - random.xlsx:	count_changed_by_approach = 12 from 53
    - 130 changed from 849

- Adson:

In [24]:
count_discordance_total = 0
count_previous_discordance_total = 0
for approach in ["approach1.xlsx", "approach2.xlsx", "approach3.xlsx", "approach4.xlsx", "approach5.xlsx", "random.xlsx"]:
    ann1_revised_dict = pd.read_excel(f"ann1/revised/{approach}", sheet_name=None)
    ann2_revised_dict = pd.read_excel(f"ann2/revised/{approach}", sheet_name=None)

    try:
        del ann1_revised_dict["templates"]
        del ann1_divergent_dict["templates"]
    except KeyError:
        pass
    count_discordance_by_approach = 0
    count_previous_discordance_by_approach = 0
    for t in ann1_revised_dict:
        try:
            discordance = ann1_revised_dict[t].classification != ann2_revised_dict[t].classification
        except ValueError as ve:
            print(approach)
            print(t)
            print("ann1")
            display(ann1_revised_dict[t].classification)
            print("ann2")
            display(ann2_revised_dict[t].classification)
            raise ve
        count_discordance_by_approach += sum(discordance)
        count_previous_discordance_by_approach += len(discordance)
    ## What we still diverge?
        # if annotator == "ann1":
        #     df_divergent = ann1_revised_dict[t][discordance]
        # elif annotator == "ann2":
        #     df_divergent = ann2_revised_dict[t][discordance]
            
    count_discordance_total += count_discordance_by_approach
    count_previous_discordance_total += count_previous_discordance_by_approach
    print(f"{approach=} ")
    print(f"divergence counting now/before: {count_discordance_by_approach}/{count_previous_discordance_by_approach}")
print(f"total: {count_discordance_total}/{count_previous_discordance_total}")
        



0